### PR TITLE
Io Kernel API stubs for logging

### PR DESCRIPTION
**Pull request by x1nixmzeng**
As per #64 these are a few more function signatures needed for logging stubs.

Please don't merge until @PatrickvL has taken a look!

I've also included some Xbox-specific structures - most have small differences from the Window variants

### CODE REVIEW

File: src/CxbxKrnl/EmuKrnlIo.cpp [hunk 1]
```diff
@@ -357,42 +376,49 @@ XBSYSAPI EXPORTNUM(71) xboxkrnl::OBJECT_TYPE xboxkrnl::IoFileObjectType =
 	'eliF' // = "File" in reverse
 };
 
-/* TODO : Finish of these Dxbx ported signatures :
 // ******************************************************************
 // * 0x0048 - IoFreeIrp()
 // ******************************************************************
-XBSYSAPI EXPORTNUM(72) xboxkrnl::NTSTATUS NTAPI xboxkrnl::IoFreeIrp
+XBSYSAPI EXPORTNUM(72) void NTAPI xboxkrnl::IoFreeIrp
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuKrnlIo.cpp]: void should be xboxkrnl::VOID

File: src/CxbxKrnl/EmuKrnlIo.cpp [hunk 2]
```diff
@@ -49,31 +49,32 @@ namespace xboxkrnl
...
 // ******************************************************************
-XBSYSAPI EXPORTNUM(60) xboxkrnl::PIRP NTAPI xboxkrnl::IoBuildAsynchronousFsdRequest
+XBSYSAPI EXPORTNUM(60) xboxkrnl::PVOID NTAPI xboxkrnl::IoBuildAsynchronousFsdRequest
 (
-	ULONG	MajorFunction,
+	IN ULONG MajorFunction,
 	PDEVICE_OBJECT	DeviceObject,
 	OUT PVOID Buffer OPTIONAL,
-	ULONG Length OPTIONAL,
+	ULONG Length,
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuKrnlIo.cpp]: this argument, and a few more in this file, should have an IN marker before it

File: src/CxbxKrnl/KernelThunk.cpp [hunk 3]
```diff
@@ -67,385 +67,385 @@ namespace xboxkrnl
...
+	(uint32)FUNC(&xboxkrnl::READ_PORT_BUFFER_UCHAR),              // 0x0149 (329)
+	(uint32)FUNC(&xboxkrnl::READ_PORT_BUFFER_USHORT),             // 0x014A (330)
+	(uint32)FUNC(&xboxkrnl::READ_PORT_BUFFER_ULONG),              // 0x014B (331)
+	(uint32)FUNC(&xboxkrnl::WRITE_PORT_BUFFER_UCHAR),             // 0x014C (332)
+	(uint32)FUNC(&xboxkrnl::WRITE_PORT_BUFFER_USHORT),            // 0x014D (333)
+	(uint32)FUNC(&xboxkrnl::WRITE_PORT_BUFFER_ULONG),             // 0x014E (334)
+	(uint32)FUNC(&xboxkrnl::XcSHAInit),                           // 0x014F (335)
+	(uint32)FUNC(&xboxkrnl::XcSHAUpdate),                         // 0x0150 (336)
+	(uint32)FUNC(&xboxkrnl::XcSHAFinal),                          // 0x0151 (337)
+	(uint32)FUNC(&xboxkrnl::XcRC4Key),					          // 0x0152 (338)
```
Comment (PatrickvL) [inline src/CxbxKrnl/KernelThunk.cpp]: this line and a few below invent the comment differently

File: src/CxbxKrnl/EmuKrnlIo.cpp [hunk 4]
```diff
@@ -597,15 +654,27 @@ XBSYSAPI EXPORTNUM(84) xboxkrnl::NTSTATUS NTAPI xboxkrnl::IoSynchronousDeviceIoC
...
+		LOG_FUNC_ARG(Length)
+		LOG_FUNC_ARG(StartingOffset)
+		LOG_FUNC_END;
 
 	LOG_UNIMPLEMENTED();
 
 	RETURN(S_OK);
 }
 
+/* TODO : Finish of these Dxbx ported signatures :
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuKrnlIo.cpp]: only three more to go until all Io kernel functions have stubs!

**Comment by PatrickvL** [comment]
Great work, almost perfect. Can be merged with or without fixing the few comments a added. Good job!

**Comment by PatrickvL** [comment]
So what would like to work on next? ;) Contact me and we'll have a chat

**Comment by LukeUsher** [comment]
Just let me know when you're ready for this to be merged and I'll do it

**Comment by x1nixmzeng** [comment]
All set, this is ready to be merged - thanks!

**Pull request merged**